File: CppPractice/Project1/Project1/Leetcode92.cpp
#include <string>

#include <vector>

#include <iostream>

#include <stdlib.h>

#include <time.h>

#include <sstream>

#include <algorithm>

#include <map>

#include <bitset>

using namespace std;


struct ListNode {
    int val;
    ListNode* next;
    ListNode() : val(0), next(nullptr) {}
    ListNode(int x) : val(x), next(nullptr) {}
    ListNode(int x, ListNode* next) : val(x), next(next) {}
};

class Solution {
public:
    ListNode* reverseBetween(ListNode* head, int left, int right) {
        vector<int> val_list;
        int list_index = 1;
        ListNode* cur = head;
        while (cur)
        {
            if (list_index>=left and list_index<=right)
            {
                val_list.push_back(cur->val);
            }
            list_index++;
            cur = cur->next;
        }
        reverse(val_list.begin(),val_list.end());
        cur = head;
        list_index = 1;
        while (cur)
        {
            if (list_index >= left and list_index <= right)
            {
                cur->val=val_list[list_index-left];
            }
            list_index++;
            cur = cur->next;

        }
        return head;

    }
};

File: Pythonleetcode/Leetcode92.py
class ListNode(object):
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next

# Val_list法
class Solution(object):
    def reverseBetween(self, head, left, right):
        """
        :type head: ListNode
        :type left: int
        :type right: int
        :rtype: ListNode
        """
        val_list=[]
        p_index=1
        p=head
        while p_index<=right:
            if p_index>=left:
                val_list.append(p.val)
            p_index+=1
            p=p.next
        val_list=val_list[::-1]
        val_index=0
        # print(val_list)
        p_index = 1
        p=head
        while p_index<=right:
            if p_index>=left:
                p.val=val_list[val_index]
                val_index+=1
            p_index+=1
            p=p.next
        return head


